### PR TITLE
Put stretch indicator behind `useMaterial3` flag

### PR DESCRIPTION
**Pull request by Piinks**
This updates the implementation of the stretching overscroll indicator to be enabled by the `useMaterial3` flag.
It deprecates the previous `androidOverscrollIndicator` property, as part of unifying M3 features and making its way to the new default. 

Related to:

- https://github.com/flutter/flutter/issues/82906
- https://github.com/flutter/flutter/issues/91605

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/app.dart [hunk 1]
```diff
@@ -782,9 +789,14 @@ class MaterialScrollBehavior extends ScrollBehavior {
   Widget buildOverscrollIndicator(BuildContext context, Widget child, ScrollableDetails details) {
     // When modifying this function, consider modifying the implementation in
     // the base class as well.
-    final AndroidOverscrollIndicator indicator = _androidOverscrollIndicator
+    late final AndroidOverscrollIndicator indicator;
+    if (Theme.of(context).useMaterial3) {
+      indicator = AndroidOverscrollIndicator.stretch;
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/app.dart:794]: This only applies to the Android platform, correct? We don't perform the overscroll on anything else?

**Review by darrenaustin — APPROVED** [review]
LGTM. Awesome Kate. I as just about to ask you what I needed to do for this 😄.

Can you also add a comment to the docs for `useMaterial3` to indicate that it turns this feature on for Android platforms? Thx.

File: packages/flutter/lib/src/material/app.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/material/app.dart:794]: That is correct!

**Comment by skia-gold** [comment]
Gold has detected about 6 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/100234

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/100234

**Pull request merged**